tools/flaky-test-retryer/github_commenter.go

```diff
@@ -35,5 +54,152 @@ func NewGithubClient(githubAccount string) (*GithubClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &GithubClient{ghc}, err
+	user, err := ghc.GetGithubUser()
+	if err != nil {
+		return nil, err
+	}
+	return &GithubClient{ghc, *user.Login}, nil
+}
+
+// PostComment posts a new comment on the PR specified in JobData, retrying the job that triggered it.
+// The comment body is dynamically built based on previous retry comments on this PR, and any old
+// comments are removed before the new one is posted.
+func (gc *GithubClient) PostComment(jd *JobData, outliers []string) error {
+	oldComment, err := gc.getOldComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number)
+	if err != nil {
+		return err
+	}
+	oldEntries, err := parseEntries(oldComment)
+	if err != nil {
+		return err
+	}
+	if _, ok := oldEntries[jd.JobName]; !ok {
+		oldEntries[jd.JobName] = 0
+	}
+	newComment, canRetry := buildNewComment(jd, oldEntries, outliers)
+	if !canRetry {
+		return fmt.Errorf("expended all %d retries", maxRetries)
+	}
+	if Flags.DryRun {
+		logWithPrefix(jd, "[dry run] Comment not updated. See it here:\n%s\n", newComment)
+		return nil
+	}
+	if oldComment != nil {
+		if err := gc.DeleteComment(jd.Refs[0].Org, jd.Refs[0].Repo, oldComment.GetID()); err != nil {
+			return err
+		}
+	}
+	_, err = gc.CreateComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number, newComment)
+	return err
+}
+
+// getOldComment queries the GitHub PR specified and gets the comment made by us. If no comment
+// is found, we do not error, since we will be creating a new one anyways.
+func (gc *GithubClient) getOldComment(org, repo string, pull int) (*github.IssueComment, error) {
+	comments, err := gc.ListComments(org, repo, pull)
+	if err != nil {
+		return nil, err
+	}
+	// this robot should only leave one comment in a given PR. Collect everything with our identifier,
+	// just in case.
+	var matches []*github.IssueComment
+	for _, comment := range comments {
+		match, err := regexp.Match(identifier, []byte(comment.GetBody()))
+		if err != nil {
+			return nil, err
+		}
+		if match && *comment.GetUser().Login == gc.Login {
+			matches = append(matches, comment)
+		}
+	}
+	// sort comments in descending order
+	sort.Slice(comments, func(i, j int) bool {
+		return comments[i].GetCreatedAt().Unix() < comments[i].GetCreatedAt().Unix()
+	})
+
+	switch len(matches) {
+	case 0:
+		return nil, nil
+	case 1:
+		return matches[0], nil
+	default:
+		if Flags.DryRun {
```

Comment:
I think relying on a global var would make unit test a bit harder, add it to `gc` might be an option

Reply:
done